### PR TITLE
fix: correct issues with modulemap generation

### PR DESCRIPTION
**Pull request by luispadron**
Fixes duplicate definition errors caused by custom/generated modulemap files.

### CODE REVIEW

File: swiftpkg/internal/swiftpkg_build_files.bzl [hunk 1]
```diff
@@ -435,7 +438,7 @@ def _clang_target_build_file(repository_ctx, pkg_ctx, target):
         noop_modulemap = clang_src_info.modulemap_path != None
         modulemap_attrs = {
             "deps": bzl_selects.to_starlark(modulemap_deps),
-            "hdrs": clang_src_info.hdrs,
+            "hdrs": [] if noop_modulemap else clang_src_info.hdrs,
```
Comment (luispadron) [inline swiftpkg/internal/swiftpkg_build_files.bzl:441]: Not needed but noticed we were adding all the hdrs here even in the noop case which seems wasteful.

**Comment by brentleyjones** [comment]
We should use an aspect hint instead (both for the empty case and the generated case): https://github.com/bazelbuild/rules_swift/blob/61e2c46dbfac6b9c11ecfb2c8e92f3e157ea9f27/swift/swift_clang_module_aspect.bzl#L221

**Comment by luispadron** [comment]
> We should use an aspect hint instead (both for the empty case and the supplied case): https://github.com/bazelbuild/rules_swift/blob/61e2c46dbfac6b9c11ecfb2c8e92f3e157ea9f27/swift/swift_clang_module_aspect.bzl#L221

hmm, this doesn't seem to work. I believe because its failing during the `objc_library (parent) -> objc_library (child)` case. 

I've tried setting the child `aspect_hints` to: `@build_bazel_rules_swift//swift:no_module` and the parent to a custom `swift_interop_hint` with the `module_map` and `module_name`. Is there an equivalent hint for `objc_library`? I dont think the interop hint does anything when its part of `deps` of another `objc_library` right?

**Comment by luispadron** [comment]
@brentleyjones I did update this to generate a `swift_interop_hint` for the parent so that we don't set `module_map` for it. For the child though I'm not sure how to get it working without setting `module_map`.

**Review by cgrindel — CHANGES_REQUESTED** [review]
Can you remove [these lines](https://github.com/cgrindel/rules_swift_package_manager/blob/3f9d430f39c057cc6925e00b93b61e9f3cd31f18/examples/firebase_example/.bazelrc#L24-L35) from the firebase example and ensure that it works? Also, can you remove [this section from the FAQ](https://github.com/cgrindel/rules_swift_package_manager/blob/3f9d430f39c057cc6925e00b93b61e9f3cd31f18/docs/faq.md?plain=1#L97-L110)?

File: swiftpkg/internal/swiftpkg_build_files.bzl [hunk 2]
```diff
@@ -1055,7 +1080,9 @@ def _new_for_license(pkg_info, license):
 
 # MARK: - Constants and API Definition
 
-swift_location = "@build_bazel_rules_swift//swift:swift.bzl"
+swift_repository = "@build_bazel_rules_swift"
+swift_location = "{rules_swift}//swift:swift.bzl".format(rules_swift = swift_repository)
+no_module_aspect_hint_label = "{rules_swift}//swift:no_module".format(rules_swift = swift_repository)
```
Comment (cgrindel) [inline swiftpkg/internal/swiftpkg_build_files.bzl:1085]: The `no_module_aspect_hint_label` does not appear to be used. Can we revert these three lines?

Comment (luispadron) [reply swiftpkg/internal/swiftpkg_build_files.bzl]: Oops leftover from testing Brentley's suggestion, removed these changes.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**